parserator_web/views.py

```diff
@@ -16,9 +17,42 @@ class AddressParse(APIView):
     def get(self, request):
         # TODO: Flesh out this method to parse an address string using the
         # parse() method and return the parsed components to the frontend.
-        return Response({})
+        address = request.query_params.get('address')
+
+        if not address:
+            return Response({
+                'data': None,
+                'error': {
+                    'message': "Missing 'address' query param "
+                }
+            }, status.HTTP_400_BAD_REQUEST)
+        else:
+            try:
+                address_components, address_type = self.parse(address)
+                return Response({
+                    'data': {
+                        'input_string': address,
+                        'address_components': address_components,
+                        'address_type': address_type
+                    },
+                    'error': None
+                })
+            except Exception as err:
+                return Response({
+                    'data': None,
+                    'error': {
+                        'message': err.__str__()
```

Comment:
You generally don't call [dundermethods](https://www.geeksforgeeks.org/dunder-magic-methods-python/) directly. Here, you would do `str(err)` instead.

Reply:
Sounds good. I'm no longer using it on this line but am elsewhere.